kore/src/Kore/Attribute/Axiom.hs

```diff
@@ -226,7 +231,7 @@ parseAxiomAttribute attr =
         Monad.>=> typed @Unit (parseAttribute attr)
         Monad.>=> typed @Idem (parseAttribute attr)
         Monad.>=> typed @Trusted (parseAttribute attr)
-        Monad.>=> typed @(Concrete Variable) (parseConcreteAttribute attr)
+        Monad.>=> typed @(Concrete Variable) (parseConcreteAttribute attr def)
```

Comment:
This is the problem: You will need to pass in the `FreeVariables` to `parseAxiomAttribute` also. Sorry, I forgot that part.